### PR TITLE
Send coverage data from TravisCI to ScurintizerCI

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

**Comment by thiemowmde** [comment]
:+1:

**Comment by thiemowmde** [comment]
:+1:

Is there an obvious way to confirm this is correct?

**Comment by JeroenDeDauw** [comment]
Look at the code coverage badge after this got merged :)

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,11 @@ install: travis_retry composer install --prefer-source
 
 script: composer ci
 
+after_success:
+  - vendor/bin/phpunit --coverage-clover coverage.clover
```
Comment (mariushoch) [inline .travis.yml]: Doesn't this mean that tests are being run twice? Also, why not restrict this to one of the PHP versions (I suggest 7 as that's the fastest).

Comment (JeroenDeDauw) [inline .travis.yml]: Like this its only run when the CI passed, and if I'm not mistaken the build and PR will be marked as success/fail without waiting on the coverage stuff, which takes quite some extra time.

Re one PHP version: sure, go for it. Personally I don't think this is worth my time though I agree it's better

Comment (thiemowmde) [inline .travis.yml]: I tried to find documentation for `after_success` but find it insufficient. I *believe* this is not run for every PHP version but only once after all PHP versions are tested.

I believe this still adds an extra run of PHPUnit on top of the six for the six different PHP versions. Is this correct? I believe this is ok. I believe creating coverage reports six times (and only using one) would be much slower than six runs without and a sevenths with coverage.

One question remains: which PHP version is used for this sevenths run?

Comment (JeroenDeDauw) [inline .travis.yml]: after_success is executed for each build AFAIK. If not, I've been incorrectly understanding that for quite some time

Comment (thiemowmde) [inline .travis.yml]: Ok. Does this mean PHPUnit will run 12 times instead of the current 6?

Comment (JeroenDeDauw) [inline .travis.yml]: yeah

**Comment by mariushoch** [comment]
Adopted so we only do the additional phpunit run on php 7.0 (once).

Untested… but I assume that this is plain bash.

**Comment by thiemowmde** [comment]
Looks good for me now. @JeroenDeDauw, I suggest you merge this now when Marius' change is ok for you.

File: .travis.yml [hunk 2]
```diff
@@ -14,6 +14,12 @@ install: travis_retry composer install --prefer-source
 
 script: composer ci
 
+after_success:
+  - if [[ "`phpenv version-name`" != "7.1" ]]; then exit 0; fi
```
Comment (JeroenDeDauw) [inline .travis.yml]: Cool, did not think of using an exit :)

I suspect this version check does not work or is at least not very robust, if this indeed gives you the full PHP version including "beta" and whatnot. You are able to get the version specified in the yaml file here somehow tho.

Comment (mariushoch) [inline .travis.yml]: This is how Travis manages the php version used… this includes the mayor + minor version, but not the release or any further appendices.

**Comment by JeroenDeDauw** [comment]
Still need to check now if it really works :)

**Pull request merged**